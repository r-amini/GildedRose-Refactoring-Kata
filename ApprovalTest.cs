using System;
using System.Collections.Generic;
using System.IO;
using System.Text;
using ApprovalTests;
using ApprovalTests.Combinations;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace csharp
{
    [UseReporter(typeof(DiffReporter))]
    [TestFixture]
    public class ApprovalTest
    {
        [Test]
        public void UpdateQuality()
        {
            CombinationApprovals.VerifyAllCombinations(
                            this.TestUpdate,
                            new string[] {
                                "Aged Brie",
                                "Backstage passes to a TAFKAL80ETC concert",
                                "Sulfuras, Hand of Ragnaros",
                                "Elixir of the Mongoose"},
                            new int[] { -1, 0, 4, 6, 8, 11 },
                            new int[] { 0, 1, 2, 50 }
                    );
        }

        private string TestUpdate(string Name, int SellIn, int Quality)
        {
            IList<Item> Items = new List<Item> { new Item { Name = Name, SellIn = SellIn, Quality = Quality } };
            GildedRose app = new GildedRose(Items);
            app.UpdateQuality();
            return Items[0].ToString();
        }
    }
}
