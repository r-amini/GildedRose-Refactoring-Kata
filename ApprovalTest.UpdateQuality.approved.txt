[Aged Brie,-1,0] => Aged Brie, -2, 2
[Aged Brie,-1,1] => Aged Brie, -2, 3
[Aged Brie,-1,2] => Aged Brie, -2, 4
[Aged Brie,-1,50] => Aged Brie, -2, 50
[Aged Brie,0,0] => Aged Brie, -1, 2
[Aged Brie,0,1] => Aged Brie, -1, 3
[Aged Brie,0,2] => Aged Brie, -1, 4
[Aged Brie,0,50] => Aged Brie, -1, 50
[Aged Brie,4,0] => Aged Brie, 3, 1
[Aged Brie,4,1] => Aged Brie, 3, 2
[Aged Brie,4,2] => Aged Brie, 3, 3
[Aged Brie,4,50] => Aged Brie, 3, 50
[Aged Brie,6,0] => Aged Brie, 5, 1
[Aged Brie,6,1] => Aged Brie, 5, 2
[Aged Brie,6,2] => Aged Brie, 5, 3
[Aged Brie,6,50] => Aged Brie, 5, 50
[Aged Brie,8,0] => Aged Brie, 7, 1
[Aged Brie,8,1] => Aged Brie, 7, 2
[Aged Brie,8,2] => Aged Brie, 7, 3
[Aged Brie,8,50] => Aged Brie, 7, 50
[Aged Brie,11,0] => Aged Brie, 10, 1
[Aged Brie,11,1] => Aged Brie, 10, 2
[Aged Brie,11,2] => Aged Brie, 10, 3
[Aged Brie,11,50] => Aged Brie, 10, 50
[Backstage passes to a TAFKAL80ETC concert,-1,0] => Backstage passes to a TAFKAL80ETC concert, -2, 0
[Backstage passes to a TAFKAL80ETC concert,-1,1] => Backstage passes to a TAFKAL80ETC concert, -2, 0
[Backstage passes to a TAFKAL80ETC concert,-1,2] => Backstage passes to a TAFKAL80ETC concert, -2, 0
[Backstage passes to a TAFKAL80ETC concert,-1,50] => Backstage passes to a TAFKAL80ETC concert, -2, 0
[Backstage passes to a TAFKAL80ETC concert,0,0] => Backstage passes to a TAFKAL80ETC concert, -1, 0
[Backstage passes to a TAFKAL80ETC concert,0,1] => Backstage passes to a TAFKAL80ETC concert, -1, 0
[Backstage passes to a TAFKAL80ETC concert,0,2] => Backstage passes to a TAFKAL80ETC concert, -1, 0
[Backstage passes to a TAFKAL80ETC concert,0,50] => Backstage passes to a TAFKAL80ETC concert, -1, 0
[Backstage passes to a TAFKAL80ETC concert,4,0] => Backstage passes to a TAFKAL80ETC concert, 3, 3
[Backstage passes to a TAFKAL80ETC concert,4,1] => Backstage passes to a TAFKAL80ETC concert, 3, 4
[Backstage passes to a TAFKAL80ETC concert,4,2] => Backstage passes to a TAFKAL80ETC concert, 3, 5
[Backstage passes to a TAFKAL80ETC concert,4,50] => Backstage passes to a TAFKAL80ETC concert, 3, 50
[Backstage passes to a TAFKAL80ETC concert,6,0] => Backstage passes to a TAFKAL80ETC concert, 5, 2
[Backstage passes to a TAFKAL80ETC concert,6,1] => Backstage passes to a TAFKAL80ETC concert, 5, 3
[Backstage passes to a TAFKAL80ETC concert,6,2] => Backstage passes to a TAFKAL80ETC concert, 5, 4
[Backstage passes to a TAFKAL80ETC concert,6,50] => Backstage passes to a TAFKAL80ETC concert, 5, 50
[Backstage passes to a TAFKAL80ETC concert,8,0] => Backstage passes to a TAFKAL80ETC concert, 7, 2
[Backstage passes to a TAFKAL80ETC concert,8,1] => Backstage passes to a TAFKAL80ETC concert, 7, 3
[Backstage passes to a TAFKAL80ETC concert,8,2] => Backstage passes to a TAFKAL80ETC concert, 7, 4
[Backstage passes to a TAFKAL80ETC concert,8,50] => Backstage passes to a TAFKAL80ETC concert, 7, 50
[Backstage passes to a TAFKAL80ETC concert,11,0] => Backstage passes to a TAFKAL80ETC concert, 10, 1
[Backstage passes to a TAFKAL80ETC concert,11,1] => Backstage passes to a TAFKAL80ETC concert, 10, 2
[Backstage passes to a TAFKAL80ETC concert,11,2] => Backstage passes to a TAFKAL80ETC concert, 10, 3
[Backstage passes to a TAFKAL80ETC concert,11,50] => Backstage passes to a TAFKAL80ETC concert, 10, 50
[Sulfuras, Hand of Ragnaros,-1,0] => Sulfuras, Hand of Ragnaros, -1, 0
[Sulfuras, Hand of Ragnaros,-1,1] => Sulfuras, Hand of Ragnaros, -1, 1
[Sulfuras, Hand of Ragnaros,-1,2] => Sulfuras, Hand of Ragnaros, -1, 2
[Sulfuras, Hand of Ragnaros,-1,50] => Sulfuras, Hand of Ragnaros, -1, 50
[Sulfuras, Hand of Ragnaros,0,0] => Sulfuras, Hand of Ragnaros, 0, 0
[Sulfuras, Hand of Ragnaros,0,1] => Sulfuras, Hand of Ragnaros, 0, 1
[Sulfuras, Hand of Ragnaros,0,2] => Sulfuras, Hand of Ragnaros, 0, 2
[Sulfuras, Hand of Ragnaros,0,50] => Sulfuras, Hand of Ragnaros, 0, 50
[Sulfuras, Hand of Ragnaros,4,0] => Sulfuras, Hand of Ragnaros, 4, 0
[Sulfuras, Hand of Ragnaros,4,1] => Sulfuras, Hand of Ragnaros, 4, 1
[Sulfuras, Hand of Ragnaros,4,2] => Sulfuras, Hand of Ragnaros, 4, 2
[Sulfuras, Hand of Ragnaros,4,50] => Sulfuras, Hand of Ragnaros, 4, 50
[Sulfuras, Hand of Ragnaros,6,0] => Sulfuras, Hand of Ragnaros, 6, 0
[Sulfuras, Hand of Ragnaros,6,1] => Sulfuras, Hand of Ragnaros, 6, 1
[Sulfuras, Hand of Ragnaros,6,2] => Sulfuras, Hand of Ragnaros, 6, 2
[Sulfuras, Hand of Ragnaros,6,50] => Sulfuras, Hand of Ragnaros, 6, 50
[Sulfuras, Hand of Ragnaros,8,0] => Sulfuras, Hand of Ragnaros, 8, 0
[Sulfuras, Hand of Ragnaros,8,1] => Sulfuras, Hand of Ragnaros, 8, 1
[Sulfuras, Hand of Ragnaros,8,2] => Sulfuras, Hand of Ragnaros, 8, 2
[Sulfuras, Hand of Ragnaros,8,50] => Sulfuras, Hand of Ragnaros, 8, 50
[Sulfuras, Hand of Ragnaros,11,0] => Sulfuras, Hand of Ragnaros, 11, 0
[Sulfuras, Hand of Ragnaros,11,1] => Sulfuras, Hand of Ragnaros, 11, 1
[Sulfuras, Hand of Ragnaros,11,2] => Sulfuras, Hand of Ragnaros, 11, 2
[Sulfuras, Hand of Ragnaros,11,50] => Sulfuras, Hand of Ragnaros, 11, 50
[Elixir of the Mongoose,-1,0] => Elixir of the Mongoose, -2, 0
[Elixir of the Mongoose,-1,1] => Elixir of the Mongoose, -2, 0
[Elixir of the Mongoose,-1,2] => Elixir of the Mongoose, -2, 0
[Elixir of the Mongoose,-1,50] => Elixir of the Mongoose, -2, 48
[Elixir of the Mongoose,0,0] => Elixir of the Mongoose, -1, 0
[Elixir of the Mongoose,0,1] => Elixir of the Mongoose, -1, 0
[Elixir of the Mongoose,0,2] => Elixir of the Mongoose, -1, 0
[Elixir of the Mongoose,0,50] => Elixir of the Mongoose, -1, 48
[Elixir of the Mongoose,4,0] => Elixir of the Mongoose, 3, 0
[Elixir of the Mongoose,4,1] => Elixir of the Mongoose, 3, 0
[Elixir of the Mongoose,4,2] => Elixir of the Mongoose, 3, 1
[Elixir of the Mongoose,4,50] => Elixir of the Mongoose, 3, 49
[Elixir of the Mongoose,6,0] => Elixir of the Mongoose, 5, 0
[Elixir of the Mongoose,6,1] => Elixir of the Mongoose, 5, 0
[Elixir of the Mongoose,6,2] => Elixir of the Mongoose, 5, 1
[Elixir of the Mongoose,6,50] => Elixir of the Mongoose, 5, 49
[Elixir of the Mongoose,8,0] => Elixir of the Mongoose, 7, 0
[Elixir of the Mongoose,8,1] => Elixir of the Mongoose, 7, 0
[Elixir of the Mongoose,8,2] => Elixir of the Mongoose, 7, 1
[Elixir of the Mongoose,8,50] => Elixir of the Mongoose, 7, 49
[Elixir of the Mongoose,11,0] => Elixir of the Mongoose, 10, 0
[Elixir of the Mongoose,11,1] => Elixir of the Mongoose, 10, 0
[Elixir of the Mongoose,11,2] => Elixir of the Mongoose, 10, 1
[Elixir of the Mongoose,11,50] => Elixir of the Mongoose, 10, 49
[Conjured,-1,0] => Conjured, -2, 0
[Conjured,-1,1] => Conjured, -2, -1
[Conjured,-1,2] => Conjured, -2, 0
[Conjured,-1,50] => Conjured, -2, 46
[Conjured,0,0] => Conjured, -1, 0
[Conjured,0,1] => Conjured, -1, -1
[Conjured,0,2] => Conjured, -1, 0
[Conjured,0,50] => Conjured, -1, 46
[Conjured,4,0] => Conjured, 3, 0
[Conjured,4,1] => Conjured, 3, -1
[Conjured,4,2] => Conjured, 3, 0
[Conjured,4,50] => Conjured, 3, 48
[Conjured,6,0] => Conjured, 5, 0
[Conjured,6,1] => Conjured, 5, -1
[Conjured,6,2] => Conjured, 5, 0
[Conjured,6,50] => Conjured, 5, 48
[Conjured,8,0] => Conjured, 7, 0
[Conjured,8,1] => Conjured, 7, -1
[Conjured,8,2] => Conjured, 7, 0
[Conjured,8,50] => Conjured, 7, 48
[Conjured,11,0] => Conjured, 10, 0
[Conjured,11,1] => Conjured, 10, -1
[Conjured,11,2] => Conjured, 10, 0
[Conjured,11,50] => Conjured, 10, 48
